### PR TITLE
Added tasks tutorial

### LINKED ISSUE #977: Add documentation for tasks
This documentation should include:

 - The difference between `DiscoveryStrategy` and `TaskManager` code.
 - How `DiscoveryStrategy` works to smooth out network traffic.
 - Using `TaskManager.register_task()`, discussing how to delay and how to loop tasks.
 - Special warning section on why not to register tasks in the `__init__` of Overlays.
 - When to use `TaskManager.register_anonymous_task()`.
 - Use `NumberCache.register_future()` for caches.

### PR DESCRIPTION
**Pull request by qstokkink**
Fixes #977

This PR:

 - Adds a tutorial on task management.
 - Adds a tutorial on `DiscoveryStrategy` creation.

These are the basics required to take #979 off of "on hold".

### CODE REVIEW

**Comment by qstokkink** [comment]
validate

**Comment by qstokkink** [comment]
validate

**Comment by qstokkink** [comment]
The remaining Flake 8 and Pylint errors are due to the documentation import system and the usage of prints. Marking this as ready.

Instead of having to compile this PR yourself, please enjoy this complementary PNG render:

taskmngr_docs

File: doc/basics/tasks_tutorial.rst [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+This example prints ``[1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]``.
+
+Note that you can also create a task with both an interval and a delay.
+
+Community Integration
+^^^^^^^^^^^^^^^^^^^^^
+
+For your convenience, every ``Community`` is also a ``TaskManager``.
+However, try to avoid spawning any tasks in the ``__init__`` of your ``Community``.
+Specifically, you could end up scheduling a task in between your ``Community`` and other ``Community`` initialization
```
Comment (devos50) [inline doc/basics/tasks_tutorial.rst:88]: This sentence is missing a period.

File: doc/basics/tasks_tutorial.rst [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+Working with a half-initialized IPv8 may lead to problems.
+This practice also makes your ``Community`` more difficult to test.
+
+If you do end up in a situation where you absolutely must schedule a task from the ``__init__()``, you can use the ``TaskManager``'s ``cancel_all_pending_tasks()`` method to cancel all registered tasks.
+This is an example of how to deal with this in your unit tests:
+
+.. literalinclude:: tasks_tutorial_6.py
+   :lines: 18-26
+
+Network IO and the DiscoveryStrategy
```
Comment (devos50) [inline doc/basics/tasks_tutorial.rst:98]: This part feels a bit detached from the overall story which is mostly about the task manager. Maybe you could move it to a separate tutorial file?

**Comment by qstokkink** [comment]
validate

File: doc/basics/discoverystrategy_tutorial.rst [hunk 3]
```diff
@@ -0,0 +1,56 @@
+Network IO and the DiscoveryStrategy
+====================================
+
+This document assumes you have a basic understanding of asyncio tasks, as documented in `the tasks tutorial <../../basics/tasks_tutorial>`_.
+You will learn how to use the IPv8's ``DiscoveryStrategy`` class to avoid network congestion.
+
+The DiscoveryStrategy
+---------------------
```
Comment (qstokkink) [inline doc/basics/discoverystrategy_tutorial.rst:8]: @devos50 These lines and the next section heading are new since your last review.

**Review by devos50 — APPROVED** [review]

**Pull request merged**